edi/models/edi_exchange_consumer_mixin.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2020 ACSONE SA
+# Copyright 2020 Creu Blanca
+# @author Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class EDIExchangeConsumerMixin(models.AbstractModel):
+    """Record that might have related EDI Exchange records
+    """
+
+    _name = "edi.exchange.consumer.mixin"
+    _description = "Abstract record where exchange records can be assigned"
+
+    exchange_record_ids = fields.One2many(
+        "edi.exchange.record",
+        inverse_name="res_id",
+        domain=lambda r: [("model", "=", r._name)],
+    )
+    exchange_record_count = fields.Integer(compute="_compute_exchange_record_count")
+
+    def _has_exchange_record(self, exchange_type, backend, extra_domain=False):
+        """This function is useful when generating the configuration"""
+        domain = [
+            ("model", "=", self._name),
+            ("res_id", "=", self.id),
+            ("backend_id", "=", backend.id),
+            ("type_id.code", "=", exchange_type),
+        ]
+        if extra_domain:
+            domain += extra_domain
+        return self.env["edi.exchange.record"].search(domain, limit=1)
```

Comment:
Even though this will work in `if self._has_exchange_record(...` I think it makes more sense to return True or False with `bool(self.env["edi...` than returning the first record it finds.

Reply:
yep, and then replace `search` w/ `search_count`